subtests/docker_cli/systemd/p4321-docker.service

```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=docker-autotest systemd subtest. If you see this file on a live system it means something has failed very badly
+After=docker.service
```

Comment:
I believe that this is the crux of the whole matter. You do not need this `After=` line. Just remove it, and all will be nice and simple again.